boskos/gcp/config.go

```diff
@@ -93,6 +99,11 @@ func (rc *resourcesConfig) Construct(res *common.Resource, types common.TypeToRe
 		return r
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), operationTimeout)
+	defer cancel()
+	errGroup, derivedCtx := errgroup.WithContext(ctx)
+	var mutex sync.RWMutex
```

Comment:
Can this name be more meaningful. Change the name to show what it is responsible for locking